File: src/main/java/models/Ticket.java
package models;

import java.sql.Date;

public class Ticket {

    private int id;
    private Integer reservationId;
    private String type;
    private double prix;
    private String statut;

    private int destinationId;
    private String destinationNom;
    private boolean selected;
    private Integer activiteId;


    public Ticket() {}

    // CREATE
    public Ticket(Integer reservationId, Integer activiteId, int destinationId,
                  String type, double prix, String statut) {
        this.reservationId = reservationId;
        this.activiteId = activiteId;
        this.destinationId = destinationId;
        this.type = type;
        this.prix = prix;
        this.statut = statut;

    }

    // READ / UPDATE
    public Ticket(int id, Integer reservationId, Integer activiteId, String type,
                  double prix, String statut,  int destinationId) {
        this.id = id;
        this.reservationId = reservationId;
        this.activiteId = activiteId;
        this.type = type;
        this.prix = prix;
        this.statut = statut;

        this.destinationId = destinationId;
    }

    public int getId() {return id;}

    public Integer getReservationId() {return reservationId;}

    public String getType() {return type;}

    public double getPrix() {return prix;}

    public String getStatut() {return statut;}



    public int getDestinationId() { return destinationId; }
    public Integer getActiviteId() { return activiteId; }
    public String getDestinationNom() { return destinationNom; }

    public boolean isSelected() { return selected;}

    public void setActiviteId(Integer activiteId) { this.activiteId = activiteId; }
    public void setId(int id) {this.id = id;}

    public void setReservationId(Integer reservationId) {this.reservationId = reservationId;}

    public void setType(String type) {this.type = type;}

    public void setPrix(double prix) {this.prix = prix;}

    public void setStatut(String statut) {this.statut = statut;}



    public void setDestinationId(int destinationId) { this.destinationId = destinationId; }

    public void setDestinationNom(String destinationNom) { this.destinationNom = destinationNom; }

    public void setSelected(boolean selected) { this.selected = selected;}
    /*@Override
    public String toString() {
        return "Ticket{" +
                "id=" + id +
                ", reservationId=" + reservationId +
                ", type='" + type + '\'' +
                ", prix=" + prix +
                ", statut='" + statut + '\'' +
                ", dateDebut=" + dateDebut +
                ", dateFin=" + dateFin +
                '}';
    }*/

    @Override
    public String toString() {
        return destinationNom + " - " + type + " - " + prix + "€";
    }
}

File: src/Entity/Ticket.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity]
class Ticket
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: "integer")]
    private ?int $id = null;

    #[ORM\Column(type: "string", length: 100)]
    #[Assert\NotBlank(message: "Le type est obligatoire")]
    #[Assert\Choice(
        choices: ["Vol", "Hotel", "Transport", "Activité"],
        message: "Type de ticket invalide"
    )]
    private string $type;

    #[ORM\Column(type: "string", length: 50, options: ["default" => "Disponible"])]
    #[Assert\NotBlank(message: "Le statut est obligatoire")]
    #[Assert\Choice(
        choices: ["Disponible", "Reservé", "Annulé"],
        message: "Statut invalide"
    )]
    private ?string $statut = null;

    #[ORM\Column(type: "float")]
    #[Assert\NotBlank(message: "Le prix est obligatoire")]
    #[Assert\Positive(message: "Le prix doit être positif")]
    #[Assert\LessThan(100000, message: "Le prix est trop élevé")]
    private float $prix;

    #[ORM\ManyToOne(targetEntity: Reservation::class, inversedBy: "tickets")]
    #[ORM\JoinColumn(name: 'reservation_id', referencedColumnName: 'id', onDelete: 'CASCADE')]
    private ?Reservation $reservation_id = null;

    #[ORM\ManyToOne(targetEntity: Ville::class)]
    #[ORM\JoinColumn(name: 'destination_id', referencedColumnName: 'id')]
    #[Assert\NotNull(message: "Veuillez sélectionner une destination")]
    private ?Ville $destination = null;

    #[ORM\ManyToOne(targetEntity: Activite::class)]
    #[ORM\JoinColumn(name: 'activite_id', referencedColumnName: 'id', onDelete: 'CASCADE')]
    private ?Activite $activite = null;

    public function __construct()
    {
        $this->statut = 'Disponible';
    }

    public function getId(): ?int { return $this->id; }

    public function getType(): ?string { return $this->type; }
    public function setType(string $value): self { $this->type = $value; return $this; }

    public function getStatut(): ?string { return $this->statut; }
    public function setStatut(?string $value): self { $this->statut = $value; return $this; }

    public function getPrix(): ?float { return $this->prix; }
    public function setPrix(float $value): self { $this->prix = $value; return $this; }

    public function getReservation_id(): ?Reservation { return $this->reservation_id; }
    public function setReservation_id(?Reservation $value): self { $this->reservation_id = $value; return $this; }

    public function getDestination(): ?Ville { return $this->destination; }
    public function setDestination(?Ville $destination): self { $this->destination = $destination; return $this; }

    public function getActivite(): ?Activite { return $this->activite; }
    public function setActivite(?Activite $activite): self { $this->activite = $activite; return $this; }
}